contracts/implementations/nodes/SyntheticDevice.sol

```diff
@@ -301,6 +309,56 @@ contract SyntheticDevice is
         _resetInfos(syntheticDeviceNode);
     }
 
+
+    /**
+     * @notice Validates burning of a synthetic device and reset all its attributes
+     * @dev Can only be called by the Synthetic Proxy when a token owner calls the `burn` function
+     * @dev The actual burn takes place on the VehicleId contract
+     * @param tokenId Synthetic Device node id
+     */
+    function validateBurnAndResetNode(uint256 tokenId) external onlyNftProxy {
+        NodesStorage.Storage storage ns = NodesStorage.getStorage();
+
+        MapperStorage.Storage storage ms = MapperStorage.getStorage();
+
+        SyntheticDeviceStorage.Storage storage sds = SyntheticDeviceStorage
+            .getStorage();
+
+        VehicleStorage.Storage storage vs = VehicleStorage
+            .getStorage();
+
+        address sdIdProxyAddress = sds.idProxyAddress;
+        address vehicleIdProxyAddress = vs.idProxyAddress;
+
+        if (!INFT(sdIdProxyAddress).exists(tokenId))
+            revert InvalidNode(vehicleIdProxyAddress, tokenId);
+
+        uint256 vehicleNode = ms.nodeLinks[sdIdProxyAddress][vehicleIdProxyAddress][
+                tokenId];
+        if (vehicleNode == 0 ) revert VehicleNotPaired(vehicleNode);
+
+        address owner = INFT(vehicleIdProxyAddress).ownerOf(tokenId);
+
+        //
+        delete ns.nodes[sdIdProxyAddress][tokenId].parentNode;
+
+        delete ms.nodeLinks[vehicleIdProxyAddress][sdIdProxyAddress][
+            vehicleNode
+        ];
+        delete ms.nodeLinks[sdIdProxyAddress][vehicleIdProxyAddress][
+            tokenId
+        ];
+
+        delete sds.deviceAddressToNodeId[
+            sds.nodeIdToDeviceAddress[tokenId]
+        ];
+        delete sds.nodeIdToDeviceAddress[tokenId];
+
+        emit SyntheticDeviceNodeBurned(tokenId, vehicleNode, owner);
```

Comment:
This is sort of awkward because the NFT hasn't been burned yet. But to be honest we don't listen for this anyway.